chaste_codegen/data/tests/chaste_reference_models/BE/li_mouse_2010.cpp

```diff
@@ -1318,63 +1305,18 @@
m_concentration__Ki)) + var_chaste_interface__cell__V) * var_chaste_interface__fast_sodium_current__O_Na * mParameters[6] - 0.25 * (-26.535875647668394 * log(mParameters[2] / var_chaste_interface__potassium_concentration__Ki) + var_chaste_interface__cell__V) * var_chaste_interface__ultra_rapidly_activating_delayed_rectifier_K_I__aur * var_chaste_interface__ultra_rapidly_activating_delayed_rectifier_K_I__iur - 0.53469999999999995 * pow(var_chaste_interface__fast_transient_outward_K_I__ato_f, 3) * (-26.535875647668394 * log(mParameters[2] / var_chaste_interface__potassium_concentration__Ki) + var_chaste_interface__cell__V) * var_chaste_interface__fast_transient_outward_K_I__ito_f - (-26.535875647668394 * log(mParameters[2] / var_chaste_interface__potassium_concentration__Ki) + var_chaste_interface__cell__V) * var_chaste_interface__non_inactivating_steady_state_K_I__aKss * mParameters[8] * var_chaste_interface__non_inactivating_steady_state_K_I__iKss - (pow(var_chaste_interface__sodium_concentration__Nai, 3) * mParameters[1] * exp(0.013189690992192795 * var_chaste_interface__cell__V) - 1000.0 * pow(mParameters[3], 3) * var_chaste_interface__calcium_concentration__Cai_converted * exp(-0.024495140414072337 * var_chaste_interface__cell__V)) * mParameters[11] / ((1.0 + 0.27000000000000002 * exp(-0.024495140414072337 * var_chaste_interface__cell__V)) * (1400.0 * pow(var_chaste_interface__sodium_concentration__Nai, 3) + 6.81472e+17 * var_chaste_interface__calcium_concentration__Cai_converted + pow(var_chaste_interface__sodium_concentration__Nai, 3) * mParameters[1] + 1728000000000.0 * (1.0 + 277.77777777777777 * var_chaste_interface__calcium_concentration__Cai_converted) * mParameters[1] + 3.6000000000000001 * pow(mParameters[3], 3) * (1.0 + 5.7870370370370368e-13 * pow(var_chaste_interface__sodium_concentration__Nai, 3)) + 1000.0 * pow(mParameters[3], 3) * var_chaste_interface__calcium_concentration__Cai_converted)) - 2.4860000000000002 * mParameters[2] / ((1.0 + 13435672565.051065 * pow((1 / var_chaste_interface__sodium_concentration__Nai), 2.3999999999999999)) * (1500.0 + mParameters[2]) * (1.0 + 0.1245 * exp(-0.0037684831406265135 * var_chaste_interface__cell__V) + 0.0052142857142857138 * (-1 + exp(1.4858841010401188e-5 * mParameters[3])) * exp(-0.037684831406265135 * var_chaste_interface__cell__V))) - 2000.0 * (40.0 + var_chaste_interface__cell__V) * var_chaste_interface__calcium_concentration__Cai_converted / ((1.0 + exp(5.9871794871794881 - 0.12820512820512822 * var_chaste_interface__cell__V)) * (10.0 + 1000.0 * var_chaste_interface__calcium_concentration__Cai_converted)) - (-26.535875647668394 * log(mParameters[2] / var_chaste_interface__potassium_concentration__Ki) + var_chaste_interface__cell__V) * mParameters[2] * mParameters[7] / ((1.0 + exp(0.089599999999999999 * var_chaste_interface__cell__V - 2.3776144580310881 * log(mParameters[2] / var_chaste_interface__potassium_concentration__Ki))) * (210.0 + mParameters[2])); // millivolt / millisecond
```

Comment:
This compression of previous 30 lines of code into 1 makes it fairly unreadable - is that needed or can we separate out again?

Reply:
Aah good point, it's caused by a new piecewise, that you don't see since it evaluated to 0 and got taken out. Maybe this special handling for when a part evaluates to 0 isn't necessary I'll test with chaste. (It also leads to state vars being taken out which looks a bit odd)

Reply:
Right so tests in chaste show that not this isn't necessary, hence I removed that bit